packages/permission-controller/src/Permission.ts

```diff
@@ -423,6 +423,12 @@ type PermissionSpecificationBase<Type extends PermissionType> = {
    * Leaving this as undefined uses default behaviour where the permission is available to request for all subject types.
    */
   subjectTypes?: readonly SubjectType[];
+
+  /**
+   * An array of associated permission names that fall under this "parent" permission. Target names would be limited to the
+   * array of child permissions associated with this particular permission.
+   */
+  children?: Readonly<TargetName[]>;
```

Comment:
Should we enforce that only a special type of permission can have children?

I am thinking it might be useful to define a `PermissionGroup` similarly to how we have endowments and RPC methods currently and only allow permissions of type "permission group" to actually function as a group.

Reply:
Re-doing this PR based on a convo w/ @rekmarks to keep a concept of groups as well, minus the need for an extraneous permission in state.